File: src/main/java/macro303/keschet/core/pieces/Scholar.java
package macro303.keschet.core.pieces;

import macro303.board_game.Colour;
import macro303.keschet.Direction;
import org.jetbrains.annotations.NotNull;

import static macro303.keschet.Direction.*;

/**
 * Created by Macro303 on 2018-02-09.
 */
public class Scholar extends Piece {
	@NotNull
//	private static final String symbol = "\uD83D\uDCDA";
	private static final String symbol = "C";
	private static final int maxDistance = 2;
	@NotNull
	private static final Direction[] validDirections = new Direction[]{NORTH, NORTH_EAST, EAST, SOUTH_EAST, SOUTH, SOUTH_WEST, WEST, NORTH_WEST};

	public Scholar(@NotNull Colour colour) {
		super(colour, maxDistance, symbol, validDirections);
	}

	public Scholar() {
		super(maxDistance, symbol, validDirections);
	}

	@Override
	public String toString() {
		return "Scholar{} " + super.toString();
	}
}

File: Base/src/main/kotlin/macro303/base/pieces/Scholar.kt
package macro303.base.pieces

import macro303.base.Colour
import macro303.base.Direction

/**
 * Created by Macro303 on 2017-10-20.
 */
class Scholar(teamColour: Colour) : Piece(
	teamColour = teamColour,
	maxDistance = 2,
	validDirections = arrayOf(
		Direction.NORTH,
		Direction.NORTH_EAST,
		Direction.EAST,
		Direction.SOUTH_EAST,
		Direction.SOUTH,
		Direction.SOUTH_WEST,
		Direction.WEST,
		Direction.NORTH_WEST
	)
) {
	override val symbol: String = Scholar.symbol

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is Scholar) return false
		if (!super.equals(other)) return false
		return true
	}

	override fun hashCode(): Int {
		return super.hashCode()
	}

	override fun toString(): String {
		return "Scholar() ${super.toString()}"
	}

	companion object {
		const val symbol = "\uD83D\uDCDA"
	}
}